src/calliope/backend/expression_parser.py

```diff
@@ -788,7 +789,7 @@ def as_array(self) -> xr.DataArray:  # noqa: D102, override
                 evaluated = backend_interface._dataset[self.name]
             except KeyError:
                 evaluated = xr.DataArray(self.name, attrs={"obj_type": "string"})
-        if "default" in evaluated.attrs:
+        if "default" in evaluated.attrs and pd.notnull(evaluated.attrs["default"]):
```

Comment:
`pd.notnull` here should change.
- Generally, `pd.isna` and `pd.notna` is preferred over `null` (https://docs.astral.sh/ruff/rules/pandas-use-of-dot-not-null/)
- This will generate odd behavior if there is a `list` in `"default"`, since it will return an object and the check will pass if there is _anything_ in it.

Reply:
Fair enough, although "default" _shouldn't_ ever have a list in it... `notnull` was always used to capture NaN and None values, but now that `notna` catches None values we may as well move to using it